### DISCUSSION TITLE
Virtualenv setup in CLion IDE

### DISCUSSION THREAD

**Discussion by rogueai** [PyO3/pyo3#3409] [Questions] [answered]
Hi all, 

I'm trying to debug my Pyo3 program in JetBrains CLion IDE, but I'm unable to have it use the local virtualenv.
Just to clarify: everything works fine if I use plain `cargo run` from terminal, but the moment I run or debug in CLion, it seems Pyo3 is only picking up the global python interpreter.

I've tried different approaches, such as setting `PYO3_PYTHON` env variable, and generating the config and passing it with `PYO3_CONFIG_FILE`, but nothing seems to work.

I understand this might very well be a CLion-specific quirk, but I wonder if there's any other way I could explicitly set the virtualenv, or some way to debug where this is going wrong?

I'm on Linux with Pyo3 version `0.19.2`

Thanks!

**Comment by rogueai** [comment, marked as answer]
I think I figured it out, not sure if this is the correct approach though, i've added this env var when launching:
```
PYTHONPATH=/path/to/project/venv/lib/python3.11/site-packages
```

I suppose this is due to the fact that I don't have an active venv when launching from CLion.

**Reply by davidhewitt** [reply]
This matches my experience in the past; I think you could also activate the venv before launching CLion.
